Sources/TuistKit/Services/AuthTokenRefreshService.swift

```diff
@@ -0,0 +1,90 @@
+import FileSystem
+import Foundation
+import Mockable
+import Path
+import ServiceContextModule
+import TuistCore
+import TuistLoader
+import TuistServer
+import TuistSupport
+
+@Mockable
+protocol AuthTokenRefreshServicing {
+    func refreshTokens(
+        path: AbsolutePath
+    ) async throws
+}
+
+struct AuthTokenRefreshService: AuthTokenRefreshServicing {
+    private let configLoader: ConfigLoading
+    private let fileSystem: FileSysteming
+    private let serverURLService: ServerURLServicing
+    private let refreshAuthTokenService: RefreshAuthTokenServicing
+    private let serverCredentialsStore: ServerCredentialsStoring
+    private let serverSessionController: ServerSessionControlling
+    private let serverAuthenticationController: ServerAuthenticationControlling
+
+    // MARK: - Init
+
+    init(
+        configLoader: ConfigLoading = ConfigLoader(warningController: WarningController.shared),
+        fileSystem: FileSysteming = FileSystem(),
+        serverURLService: ServerURLServicing = ServerURLService(),
+        refreshAuthTokenService: RefreshAuthTokenServicing = RefreshAuthTokenService(),
+        serverCredentialsStore: ServerCredentialsStoring = ServerCredentialsStore(),
+        serverSessionController: ServerSessionControlling = ServerSessionController(),
+        serverAuthenticationController: ServerAuthenticationControlling = ServerAuthenticationController()
+    ) {
+        self.configLoader = configLoader
+        self.fileSystem = fileSystem
+        self.serverURLService = serverURLService
+        self.refreshAuthTokenService = refreshAuthTokenService
+        self.serverCredentialsStore = serverCredentialsStore
+        self.serverSessionController = serverSessionController
+        self.serverAuthenticationController = serverAuthenticationController
+    }
+
+    func refreshTokens(
+        path: AbsolutePath
+    ) async throws {
+        let config = try await configLoader.loadConfig(path: path)
+        let serverURL = try serverURLService.url(configServerURL: config.url)
```

Comment:
```suggestion
        serverURL: URL
    ) async throws {
```

This service only needs to know the `serverURL`. That information should be passed from the command service, so we don't load the config multiple times.